### ISSUE TITLE
Tablature input: alphanumeric keyboard digits no longer work (relative to the AZERTY keyboard?)

### ISSUE THREAD

**Issue by cadiz2** [musescore/MuseScore#20741]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

See title and steps to reproduce below

### Steps to reproduce

1. Guitar score, TAB staff
2. Enter numbers 1, 2, 3, etc. 
Result: nothing happens. You need to add the Shift key, or type the letters A, B, C etc. to achieve this. This is a regression compared to other versions 2 and 3, where numbers input worked perfectly well with the alphanumeric keyboard digits (and WITHOUT THE SHIFT KEY). Which is perfectly annoying and cumbersome, so, I continue using the V3

NB: in the GIF below, you'll notice that 1/& and 2/é don't even appear in AutoHotkey. From 3/" onwards only. And you'll also notice that the right arrow key jumps from one measure to the next. 

I issued another bug report a year ago, see: https://github.com/musescore/MuseScore/issues/15204#

### Screenshots/Screen recordings

See below (additional context). Sorry. 

### MuseScore Version

Version 4.2

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

Windows10

### Additional context

![Video_2023-12-28_144253](https://github.com/musescore/MuseScore/assets/114975887/957fa071-5875-4cc7-975d-9e30327ee82d)

**Comment by MarcSabatella** [comment]
Did you reset your shortcuts since initially installing the original 4.0?  The bug was definitely fixed in the defaults shortcut file long ago, but if you had previously customized your shortcuts, then you'd still have the incorrect settings for those commands.  Best to try Help / Revert to factory settings.

**Comment by cadiz2** [comment]
And of course, in 1 year, I've had the opportunity to Revert to Factory Setting a few times, and again just now. It doesn't change a thing.

**Comment by MarcSabatella** [comment]
I orignally thought you were talking about something different - a known long-ago-fixed issues with the numeric keypad - but now I am seeing it different.  To make sure I understand: you are talking about the standard numeric keys on your keyboard and *not* the numeric keypad?  On AZERTY, these are normally located *above* other keys, and do require Shift:

![image](https://github.com/musescore/MuseScore/assets/1799936/7f8da9f8-8b07-42ad-8497-d8815113e4d5)

So, you're saying, you aren't actually entering a number, but are entering one of those other keys, and wanting it to act *as if* you had entered the number?  So for instance, typing "-" would not give you a flat as it normally would, but instead enter a 6?

It seems it would be possible to hand-edit the shortcuts.xml file to produce this result.  Would you be expecting this result only for tablature?  Seems that giving up the normal behavior of those other keys would not be a good thing in general, but should be harmless enough in tablature entry.  So I could imagine this being incorporated into the standard release.

**Comment by cadiz2** [comment]
Indeed, I talk about the standard numeric keys (above the letters) since it's the way to input Tablatures, handbook "Press a number from 0 to 9 to enter a fret mark from 0 to 9"
Numeric keyboard is dedicated for selecting notes values, as always too. 

*"So, you're saying, you aren't actually entering a number, but are entering one of those other keys, and wanting it to act as if you had entered the number? So for instance, typing "-" would not give you a flat as it normally would, but instead enter a 6? And "(" wouldn't add parentheses, but would enter a 5?"*

Absolutely, as I've always done with versions 2 and 3

**Comment by MarcSabatella** [comment]
Would you want that capability to be *in addition* to the numbers working normally (with Shift), or *instead*?  The latter you can do easily enough via Edit / Preferences / Shortcuts.  But to have both sets of shortcuts available, you would need to hand-editing the shortcuts file.  That's pretty simple, though.  If this is something you think other French users might also be interested in, probably someone has already done this and could share their custom shortcuts file.  Have you asked on the forum?

Here is the process:

1) Go to Edit / Preferences / Shortcuts
2) click Export
3) save XML file to your computer
4) edit that XML file (see below)
5) in Edit / Preferences / Shortcuts, click Import
6) select the edited XML file

The actual edit is a matter of finding the entries for fret-1 through fret-9 and adding lines of the form `<seq>-</seq>`.  A complication is that a bunch of those symbols probably need to be "escaped" so they don't get interpreted as XML.  And not having an AZERTY keyboard, I would have no way of testing this.  Which is why it would be nice to find someone who has already done it.

**Comment by cadiz2** [comment]
Here's what I see in the V4 shortcuts - see image below.

You say: "_The actual edit is a matter of finding the entries for fret-1 through fret-9 and adding lines of the form <seq>-</seq>_
Could you be more specific, and give me an example or two of what I need to change. 
For example: Fret 1: currently: 1;B
What do I need to add or modify to this shortcut? Where is it supposed to be " <seq>-</seq>"
Could you write it down concretely? Thanks. 

![shorcuts](https://github.com/musescore/MuseScore/assets/114975887/78c2ce1f-855c-44ac-80f4-03c8d78ee303)

**Comment by cadiz2** [comment]
For the record, the same shorcuts with the V3. 

![shortcuts v3](https://github.com/musescore/MuseScore/assets/114975887/81bf5877-a78c-4a7d-975f-ad55982a4751)

**Comment by cadiz2** [comment]
I tried to modify the first three shortcuts 0, 1, and 2 by copying those of V3. But in the text editor (Notepad++), I can't get the accented A, ampersand and accented E to capitalize - image below. And when I import this modified file, it's as if nothing had happened: the "default" shortcuts of V4 are still the same. 

![shorcuts 3](https://github.com/musescore/MuseScore/assets/114975887/9b9689bc-18d6-48ef-b2b5-eba88f8db38f)

**Comment by MarcSabatella** [comment]
Let me try to explain again:

If you are OK with using those special characters *instead* of the actual numbers, you don't need to export or import or hand-edit the XML file at all.  Just go to Edit / Preferences / Shortcuts, find and select the command you want to customize, then click Define, then type the character you want to use *instead* of the default.

If you want to use the special characters *in addition* to being able to use numbers or letters, then you need to edit the XML file.  But as I also said, many of those characters are special to XML and probably need to be "escaped" in order to be recognized.  That means things like using `&amp;` instead of `&` and so forth.  And I have no way of testing how that would work for those characters on AZERTY.  So better to ask on the forum to see if other French users have done this.

**Comment by cadiz2** [comment]
I want to use the special characters _in addition_ to being able to use numbers or letters. 
Strangely, I haven't seen any such complaint on the French forum, even though the shortcuts have changed in the same way between the two versions. I can't be the only one who wants numbers on tablatures though! Maybe there are users who use the AZERTY keyboard even on the English-speaking forum!
Otherwise, I simply tried to import the xml file from V3 to V4. Numbers and letters now work fine for Tablatures. But I notice, for example (I not verify all) , that the Tie  shortcut no longer works, neither "+" nor "T" (I don't know why).
I'll see how to keep going. Thanks for helping.

**Comment by cadiz2** [comment]
Solved. By editing the XML file.
